clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -261,6 +262,20 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
         uint64_t last_run_time = 0;
     };
 
+    // XXX i#6831: Should this live entirely inside the dynamic subclass?
+    // We would need to make part of ini() virtual with subclass overrides.
+    struct workload_info_t {
+        explicit workload_info_t(int output_limit)
+            : output_limit(output_limit)
+        {
+            live_outputs = std::unique_ptr<std::atomic<int>>(new std::atomic<int>());
+            live_outputs->store(0, std::memory_order_relaxed);
+        }
+        int output_limit; // Read-only after constructed.
+        std::unique_ptr<std::atomic<int>> live_outputs;
+        std::vector<input_ordinal_t> inputs; // Read-only after constructed.
```

Comment:
Not really read-only after constructed.

Can't we accept it in the constructor and make it const as well?

Reply:
The comment is about whether we need a lock to read this: we don't since it's not written by any thread after a single thread constructs it during initialization.  Changed the comment to `// No lock needed: read-only post-init.`